test/src/memory_review_settings_source_test.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:review_service/src/review_service/review_settings.dart';
+import 'package:review_service/src/review_service/memory_review_settings_source.dart';
+
+void main() {
+  test('MemoryReviewSettingSource should not return null when reading from a new instance', () async {
+    // Arrange
+		var source = new MemoryReviewSettingsSource<ReviewSettings>();
+
+		// Act
+		var result = await source.read();
+
+		// Assert
+    expect(result, isNotNull);
+  });
+  test('MemoryReviewSettingSource should return the same value that was written when reading', () async {
+    // Arrange
+    var source = MemoryReviewSettingsSource<ReviewSettings>();
+    var value = ReviewSettings(requestCount: 10);
+
+    await source.write(value);
+
+    // Act
+    var result = await source.read();
+
+    // Assert
+    expect(value, result);
```

Comment:
I think you want the first parameter to be the actual result and the second parameter to be the expected result, so if it fails, the error message makes sense